delivery_carrier_package_measure_required/models/stock_quant_package.py

```diff
@@ -8,20 +8,23 @@
 class StockQuantPackage(models.Model):
     _inherit = "stock.quant.package"
 
-    length_required = fields.Boolean(related="packaging_id.package_length_required")
-    width_required = fields.Boolean(related="packaging_id.package_width_required")
-    height_required = fields.Boolean(related="packaging_id.package_height_required")
-    weight_required = fields.Boolean(related="packaging_id.package_weight_required")
+    length_required = fields.Boolean(related="package_type_id.package_length_required")
+    width_required = fields.Boolean(related="package_type_id.package_width_required")
+    height_required = fields.Boolean(related="package_type_id.package_height_required")
+    weight_required = fields.Boolean(related="package_type_id.package_weight_required")
 
     # The boolean field use to check if a dimension is required are intentionally left out.
     # To not raise error when changing packaging configuration.
     @api.constrains("pack_length", "width", "height", "shipping_weight", "quant_ids")
     def _check_required_dimension(self):
         ignore_pack_content = self.env.context.get(
-            "delivery_pkg_measure__ignore_package_content"
+            "delivery_pkg_measure__ignore_package_content", False
+        )
+        force_validation = self.env.context.get(
+            "delivery_pkg_measure__force_validation_package", False
         )
         for package in self:
-            if not ignore_pack_content and not package.quant_ids:
+            if ignore_pack_content or (not force_validation and not package.quant_ids):
```

Comment:
@jbaudoux , I think this is better now.

Before the changes we don't check the dimensions in the case if ignore_pack_content is False and the package has no quants.
We also have a variable that is passed by context which is delivery_pkg_measure__ignore_package_content that if you pass it with the value set to True you would expect it not to check the dimensions but this is not the case.

Option1:
ignore_pack_content = True, package.quant_ids = True
(not ignore_pack_content and not package.quant_ids) if == False --> Check dimensions
Opción 2:
ignore_pack_content = True, package.quant_ids = False
(not ignore_pack_content and not package.quant_ids) if == False --> Check dimensions
Opción 3:
ignore_pack_content = False, package.quant_ids = True
(not ignore_pack_content and not package.quant_ids) if == False --> Check dimensions
Opción 4:
ignore_pack_content = False, package.quant_ids = False
(not ignore_pack_content and not package.quant_ids) if == True --> No Check dimensions

But what if we want to be able to skip the dimension check in other cases?

Option1:
ignore_pack_content = False, force_validation = True, package.quant_ids = True
ignore_pack_content or (not force_validation and not package.quant_ids) if = False --> Check dimensions
Opción 2:
ignore_pack_content = False, force_validation = True, package.quant_ids = False
ignore_pack_content or (not force_validation and not package.quant_ids) if = False --> Check dimensions
Opción 3:
ignore_pack_content = False, force_validation = False, package.quant_ids = True
ignore_pack_content or (not force_validation and not package.quant_ids) if = False --> Check dimensions
Opción 4:
ignore_pack_content = False, force_validation = False, package.quant_ids = False
ignore_pack_content or (not force_validation and not package.quant_ids) if = True --> No Check dimensions
Opción 5:
ignore_pack_content = True
ignore_pack_content or (not force_validation and not package.quant_ids) if = True --> No Check dimensions

What do you think about this? this can go in another commit if you think it is more correct.

Reply:
however, the delivery_pkg_measure__ignore_package_content context I have not seen used in other OCA modules.

Reply:
I understand @jbaudoux , but now I have client developments that need these contexts. We can change this "if ignore_pack_content or (not force_validation and not package.quant_ids):" to "if self.ignore_check_dimensions():" and have everyone inherit the method to whatever we need.

Reply:
ping @TDu

Reply:
As commented in the code that content key `delivery_pkg_measure__ignore_package_content`  was meant to be used to force asking for dimension even for empty packages.

I think the idea behind it was, you receive a new product in your warehouse, all packaging (A, B) for that product are not configured properly. The product arrives in packaging A. But you want the user to configure all packaging (A and B) for that product on first reception.

Now indeed it does not look like it is used :think

Reply:
Is it a problem to accept this new proposal? Do you need it to be in a separate commit?

Reply:
Done @jbaudoux, sorry for the delay